core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -54,9 +66,35 @@
   private Executor executor;
   private final List<ClientInterceptor> interceptors = new ArrayList<ClientInterceptor>();
 
+  private final URI target;
+
+  @Nullable
+  private final SocketAddress directServerAddress;
+
   @Nullable
   private String userAgent;
 
+  @Nullable
+  private NameResolver nameResolver;
+
+  @Nullable
+  private LoadBalancer loadBalancer;
+
+  protected AbstractManagedChannelImplBuilder(URI target) {
+    this.target = Preconditions.checkNotNull(target);
+    this.directServerAddress = null;
+  }
+
+  protected AbstractManagedChannelImplBuilder(SocketAddress directServerAddress) {
```

Comment:
If we do so, we probably need two schemes for in-process, one for InProcessTransport, the other for netty's LocalChannel. Looks InProcessTransport connection is bound solely by the name. I am not sure LocalChannel is the case, or it's bound by the same LocalAddress object.